development/gcp/cloud-functions/rotateserviceaccount/rotateserviceaccount.go

```diff
@@ -0,0 +1,125 @@
+package rotateserviceaccount
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+
+	"cloud.google.com/go/compute/metadata"
+	"github.com/kyma-project/test-infra/development/gcp/pkg/cloudfunctions"
+	"github.com/kyma-project/test-infra/development/gcp/pkg/pubsub"
+	"github.com/kyma-project/test-infra/development/gcp/pkg/secretmanager"
+	"google.golang.org/api/iam/v1"
+)
+
+var (
+	projectID             string
+	secretManagerService  *secretmanager.Service
+	serviceAccountService *iam.Service
+)
+
+// ServiceAccountJSON stores Service Account athentication data
+type ServiceAccountJSON struct {
+	Type             string `json:"type"`
+	ProjectID        string `json:"project_id"`
+	PrivatekayID     string `json:"private_key_id"`
+	PrivateKey       string `json:"private_key"`
+	ClientEmail      string `json:"client_email"`
+	ClientID         string `json:"client_id"`
+	AuthURL          string `json:"auth_uri"`
+	TokenURI         string `json:"token_uri"`
+	AuthProviderCert string `json:"auth_provider_x509_cert_url"`
+	ClientCert       string `json:"client_x509_cert_url"`
+}
+
+func init() {
+	var err error
+	ctx := context.Background()
+
+	projectID, err = metadata.ProjectID()
+	if err != nil {
+		panic("failed to retrieve GCP Project ID, error: " + err.Error())
+	}
+
+	secretManagerService, err = secretmanager.NewService(ctx)
+	if err != nil {
+		panic("failed creating Secret Manager client, error: " + err.Error())
+	}
+
+	serviceAccountService, err = iam.NewService(ctx)
+	if err != nil {
+		panic("failed creating IAM client, error: " + err.Error())
+	}
+}
+
+// RotateServiceAccount adds new secret version in Secret Manager on pubsub rotate message
+func RotateServiceAccount(ctx context.Context, m pubsub.MessagePayload) error {
+	var err error
+	var secretRotateMessage pubsub.SecretRotateMessage
+	var secretData ServiceAccountJSON
+
+	// Create logger to use google cloud functions structured logging
+	logger := cloudfunctions.NewLogger()
+	// Set component for log entries to identify all messages for this function.
+	logger.WithComponent("kyma.prow.cloud-function.RotateServiceAccount")
+	// Set trace value for log entries to identify messages from one function call.
+	logger.GenerateTraceValue(projectID, "RotateServiceAccount")
+
+	if m.Attributes["eventType"] != "SECRET_ROTATE" {
+		logger.LogDebug("Unsupported event type: %s, quitting", m.Attributes["eventType"])
+		return nil
+	}
+
+	err = json.Unmarshal(m.Data, &secretRotateMessage)
+	if err != nil {
+		logger.LogCritical("failed to unmarshal message data field, error: %s", err.Error())
+	}
+
+	if secretRotateMessage.Labels["type"] != "service-account" {
+		logger.LogDebug("Unsupported secret type: %s, quitting", secretRotateMessage.Labels["type"])
```

Comment:
I would log this as Info severity. This message is not about function internal state or decision. It's about stopping whole execution because of input data doesn't meet requirements to process it.
```suggestion
		logger.LogInfo("Unsupported secret type: %s, quitting", secretRotateMessage.Labels["type"])
```

Reply:
There is no need to notify on production when the event is unsupported. It's indeed stopping the execution of the program, but it's an inner decision to stop on unsupported type. This can simply be a debug and on prod silently exit to not generate any not meaningful logs and not litter Stackdriver, given we might have hundreds pubs in messages in the future